File: Teorver/2/main.cpp
#include <iostream>
#include <cmath>
#include <iomanip>
using std::cin;

int factorial(int i)
{
  if (i==0) return 1;
  else return i*factorial(i-1);
}

void first_task(long double lambda, long double nu){
    long double sum = 0, M = 0, k = 0, ro = lambda/nu;
    for (int i = 0; i < 4; i++){
        long double p_0 = 1/(sum + pow(ro, i)/factorial(i));
        sum += p_0;
        long double p = pow(ro, i)/factorial(i) * p_0;
        M += p * i;
        std::cout << "n = " << i << " p = " << p << " M = " << M;
        if (i > 0) std::cout << " k = " << M/i;
        std::cout << '\n';
    }
}

int main(){
    int R1, G1, B1, R2, G2, B2, R3, G3, B3;
    cin >> R1 >> G1 >> B1;
    cin >> R2 >> G2 >> B2;
    cin >> R3 >> G3 >> B3;

    long double T_c = R1 + G1 + B1, T_s = R2;
    long double lambda = 1/T_c, nu = 1/T_s;

    first_task(lambda, nu);
}

File: Teorver/2/main.py
import math

def first_task(lamb, nu):
    local_p = []
    ro = lamb/nu
    sum = 0
    n = 0
    while(True):
        sum += (ro**n)/math.factorial(n)
        local_p.append((ro**n)/math.factorial(n))
        if ((ro**n)/math.factorial(n) < 0.01):
            break
        n+=1
    p0 = 1/sum
    local_p = [local_p * p0 for local_p in local_p]
    N = 0
    arr_N = []
    for n in range(len(local_p)):
        N += n * local_p[n]
        arr_N.append(N)
    for n in range(len(local_p)):
        if (n == 0): continue
        print(f'p{n} = {local_p[n]}')
        print(f'N{n} = {arr_N[n]}')
        print(f'k{n} = {N/n}')
        print('\n')
    return n

def second_task(lamb, nu, n): # NOT FINISHED
    local_p = []
    ro = lamb/nu
    sum = 0
    n = 0
    for i in range(n):
        sum += (ro**i)/math.factorial(i)
        local_p.append((ro**i)/math.factorial(i))

    return n

lamb = 1/19
nu = 1/7
a = first_task(lamb, nu)
second_task(lamb, nu, a)